common/metrics/statsd_exporter.go

```diff
@@ -0,0 +1,287 @@
+			return err
+		}
+	}
+	return nil
+}
+
+func (e *statsdExporter) exportHistogramInt64(name string, data metricdata.Histogram[int64]) error {
+	for _, dp := range data.DataPoints {
+		metricName := e.buildMetricName(name, dp.Attributes)
+
+		// Export histogram as multiple metrics
+		// Count
+		if err := e.client.Inc(metricName+".count", int64(dp.Count), 1.0); err != nil {
+			return err
+		}
+
+		// Sum (dp.Sum is just an int64, not a pointer or optional type)
+		if err := e.client.Gauge(metricName+".sum", dp.Sum, 1.0); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (e *statsdExporter) exportHistogramFloat64(name string, data metricdata.Histogram[float64]) error {
+	for _, dp := range data.DataPoints {
+		metricName := e.buildMetricName(name, dp.Attributes)
+
+		// Export histogram as multiple metrics
+		// Count
+		if err := e.client.Inc(metricName+".count", int64(dp.Count), 1.0); err != nil {
+			return err
+		}
+
+		// Sum (dp.Sum is just a float64, not a pointer or optional type)
+		if err := e.client.Gauge(metricName+".sum", int64(dp.Sum), 1.0); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (e *statsdExporter) buildMetricName(name string, attrs attribute.Set) string {
+	if attrs.Len() == 0 {
+		return name
+	}
+	tags := e.buildTags(attrs)
+	// if a tag separator is provided, we need to emit the tags separately.
+	if e.tagSeparator != "" {
+		return appendSeparatedTags(name, e.tagSeparator, tags)
+	}
+
+	// if no tag separator is provided, we need to embed the tags in the metric name.
+	return embedTags(name, tags)
+}
+
+// embedTags adds the sorted list of tags directly in the stat name.
+// For example, if the stat is `hello.world` and the tags are `{universe: milkyWay, planet: earth}`,
+// the stat will be emitted as `hello.world.planet.earth.universe.milkyWay`.
+func embedTags(name string, tags []statsd.Tag) string {
+	var buffer strings.Builder
+	buffer.WriteString(name)
+	for _, tg := range tags {
+		// adding "." as delimiter so that it will show as different parts in Graphite/Grafana
+		buffer.WriteString("." + tg[0] + "." + tg[1])
+	}
+
+	return buffer.String()
+}
+
+// appendSeparatedTags adds the sorted list of tags using the DogStatsd/InfluxDB supported tagging protocol.
+// For example, if the stat is `hello.world` and the tags are `{universe: milkyWay, planet: earth}` and the separator is `,`,
+// the stat will be emitted as `hello.world,planet=earth,universe=milkyWay`.
+//
+// For more details on the protocol see:
+// - Datadog: https://docs.datadoghq.com/developers/dogstatsd/datagram_shell
+// - InfluxDB: https://github.com/influxdata/telegraf/blob/ce9411343076b56dabd77fc8845cc58872d4b2e6/plugins/inputs/statsd/README.md#influx-statsd
+func appendSeparatedTags(name string, separator string, tags []statsd.Tag) string {
+	var buffer strings.Builder
+	buffer.WriteString(name)
+	for _, tg := range tags {
+		buffer.WriteString(separator + tg[0] + "=" + tg[1])
+	}
+	return buffer.String()
+}
```

Comment:
This is copied from our existing implementation of statsd reporter for tally - 
 https://github.com/temporalio/temporal/blob/main/common/metrics/tally/statsd/reporter.go#L91-L92